File: Bronze1-1/06-23-am/unit07/taming/taming.cpp
/*
 * Taming (Feb 2018)
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	ifstream cin("taming.in");
	ofstream cout("taming.out");
	
	int N; cin >> N ;   //number of days
	int A[N];   //the log of records
	for(int k=0; k<N; k++)
		cin >> A[k];

	bool consistant = true;
	A[0] = 0;    //mark day 0 as accident day
	for(int j=N-2; j>=0; j--) {
		if( A[j+1]>0 ) {
			if( A[j]==-1 )
				A[j] = A[j+1]-1;
			else if( A[j]!=A[j+1]-1 ) {
				consistant = false;
				break;
			}
		}
	}
	if( !consistant ) {
		cout << -1 << endl;
		return 0;
	}
	
	//find the minimum number of days with accident
	int m = 0;
	for(int j=0; j<N; j++)
		if(A[j]==0)
			m++;
	
	//find the maximum number of days with accident
	int M = m;
	for(int j=0; j<N; j++)
		if(A[j]==-1)
			M++;
	
	cout << m << " " << M << endl;
}

File: Bronze1-1/06-23-am/taming.java
/*
 * Taming (Feb 2018)
 */
import java.io.*;
import java.util.*;
public class taming {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(new File("taming.in"));
		PrintWriter out = new PrintWriter(new File("taming.out"));
		
		int N=in.nextInt();   //number of days
		int[] A=new int[N];   //the records
		for(int k=0; k<N; k++)
			A[k]=in.nextInt();

		boolean consistent = true;
		A[0] = 0;  //mark day 0 as accident day
		for(int j=N-2; j>=0; j--) {
			if(A[j+1]>0) { //if next day value v>0, then today's value is v-1
				if(A[j]==-1)
					A[j] = A[j+1]-1;
				
				else if(A[j]!=A[j+1]-1) {
					consistent = false;
					break;
				}
			}
		}
		
		if(consistent) {
			int min = 0; //minimum number of days with accident
			for(int j=0; j<N; j++)
				if(A[j]==0)
					min++;
			
			int max = min; //maximum number of days with accident
			for(int j=0; j<N; j++)
				if(A[j]==-1)
					max++;
			
			out.println(min + " " + max);
		}
		else 
			out.println(-1);
		out.close();
		in.close();
	}
}
